h/static/scripts/controllers/form-controller.js

```diff
@@ -243,7 +243,7 @@ class FormController extends Controller {
         // If there was an error processing the request or the server could
         // not be reached, display a helpful error
         this.setState({
-          submitError: err.reason,
+          submitError: err.reason || 'Form Processing Error',
```

Comment:
I think we should tweak the wording here to sound a little more friendly. Perhaps "There was a problem saving changes."?